etw/java/src/main/java/com/microsoft/applicationinsights/agentc/internal/diagnostics/etw/events/model/IpaEtwEventErrorBase.java

```diff
@@ -0,0 +1,47 @@
+/*
+ * ApplicationInsights-Java
+ * Copyright (c) Microsoft Corporation
+ * All rights reserved.
+ *
+ * MIT License
+ * Permission is hereby granted, free of charge, to any person obtaining a copy of this
+ * software and associated documentation files (the ""Software""), to deal in the Software
+ * without restriction, including without limitation the rights to use, copy, modify, merge,
+ * publish, distribute, sublicense, and/or sell copies of the Software, and to permit
+ * persons to whom the Software is furnished to do so, subject to the following conditions:
+ * The above copyright notice and this permission notice shall be included in all copies or
+ * substantial portions of the Software.
+ * THE SOFTWARE IS PROVIDED *AS IS*, WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED,
+ * INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR
+ * PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE
+ * FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR
+ * OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
+ * DEALINGS IN THE SOFTWARE.
+ */
+package com.microsoft.applicationinsights.agentc.internal.diagnostics.etw.events.model;
+
+public abstract class IpaEtwEventErrorBase extends IpaEtwEventBase {
+    private StacktraceProvider stacktrace;

```

Comment:
Is it important that this has no dependencies? Wondering if we need the `StacktraceProvider` abstraction, or can we depend directly on `IThrowableProxy` here?

Reply:
This was the only dependency on logback. I used to use `IThrowableProxy` and I changed it in commit addb62921ba337b49e076781ae56483fbf9bcbbf, but I don't recall exactly what broke when I fixed the circular dependency. I can backtrack if this becomes an issue.

Reply:
Oh ya, I had missed before that this class is in `etw:java` 👍